File: src/main/java/com/header/header/domain/menu/entity/Menu.java
package com.header.header.domain.menu.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Table(name = "tbl_menu")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@ToString
public class Menu {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer menuCode;
    private String menuName;
    private Integer menuPrice;
    private Integer estTime;
    private Boolean isActive;

    /**
     * 메뉴카테고리와의 연관관계
     *
     * @JoinColumns: 복합키를 참조하기 위해 2개 컬럼 조인
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumns({
        @JoinColumn(name = "category_code", referencedColumnName = "categoryCode"),
        @JoinColumn(name = "shop_code", referencedColumnName = "shopCode")
    })
    @ToString.Exclude   // 연관 엔티티 무한 순환 방지
    private MenuCategory menuCategory;

    // 업데이트 메소드들
    public void updateMenuInfo(String menuName, Integer menuPrice, Integer estTime) {
        this.menuName = menuName;
        this.menuPrice = menuPrice;
        this.estTime = estTime;
    }

    public void updateActiveStatus(Boolean isActive) {
        this.isActive = isActive;
    }

}

File: llm/app/models/menu.py
# app/models/menu.py
from sqlalchemy import Table, Column, Integer, String, Boolean
from app.core.db import metadata

Menu = Table(
    'tbl_menu',
    metadata,
    Column('menu_code', Integer, primary_key=True, autoincrement=True),
    Column('category_code', Integer, nullable=False, comment='카테고리 코드'),
    Column('shop_code', Integer, nullable=False, comment='샵 코드'),
    Column('menu_name', String(40), nullable=False, comment='시술명'),
    Column('menu_price', Integer, nullable=False, comment='시술가격'),
    Column('est_time', Integer, nullable=False, comment='예상소요시간'),
    Column('is_active', Boolean, nullable=False, default=True, comment='활성 여부')
)
